unity-client/Assets/Scripts/MainScripts/DCL/Controllers/Scene/SceneUtils.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+using DCL.Components;
+using DCL.Models;
+
+namespace DCL.Controllers
+{
+    public static class SceneUtils
+    {
+        public static DecentralandEntity DuplicateEntity(ParcelScene scene, DecentralandEntity entity)
+        {
+            if (!scene.entities.ContainsKey(entity.entityId)) return null;
+
+            DecentralandEntity newEntity = scene.CreateEntity(System.Guid.NewGuid().ToString());
+
+            if (entity.children.Count > 0)
+            {
+                using (var iterator = entity.children.GetEnumerator())
+                {
+                    while (iterator.MoveNext())
+                    {
+                        DecentralandEntity childDuplicate = DuplicateEntity(scene, iterator.Current.Value);
+                        childDuplicate.SetParent(newEntity);
+                    }
+                }
+            }
+
+            if (entity.parent != null)
+                scene.SetEntityParent(newEntity.entityId, entity.parent.entityId);
+
+            DCLTransform.model.position = WorldStateUtils.ConvertUnityToScenePosition(entity.gameObject.transform.position);
+            DCLTransform.model.rotation = entity.gameObject.transform.rotation;
+            DCLTransform.model.scale = entity.gameObject.transform.lossyScale;
+
+            foreach (KeyValuePair<CLASS_ID_COMPONENT, IEntityComponent> component in entity.components)
+            {
+                scene.EntityComponentCreateOrUpdateWithModel(newEntity.entityId, component.Key, DCLTransform.model);
```

Comment:
Shouldnt we use the component's model instead of `DCLTransform.model` here?

Reply:
I think so, but this code was already here. @Suduck maybe can enlighten us.

Reply:
Yes we should use the model here but we should have a special case for `DCLTransform `since the component doesn't have a model that herits from `BaseModel`

Reply:
Sorry after the review I saw that now the DCLTransform heir from baseModel, we can just change it!

Just to give more context, in the builder in world the only component that we are using (right now) as `BaseComponent` is Transform, that was the reason that this was working